src/boards/rainbow_esp.h

```diff
@@ -18,6 +18,7 @@
 
 static uint8 const NO_WORKING_FILE = 0xff;
 static uint8 const NUM_FILE_PATHS = 3;
+static uint8 const NUM_FILES = 9;
```

Comment:
You should use it in declaration of `BrokeStudioFirmware::files` and `BrokeStudioFirmware::file_exists` replacing the hardcoded `64` there.

Also Rainbow's documentation states that there are 64 files. See `FILE_OPEN`, `FILE_EXISTS` and `FILE_DELETE` description of field `File index`.